### PR TITLE
Add_dress

### PR DESCRIPTION
**Pull request by GMart818**
This is code added to show sign up/in, adding an item, and checking out.

### CODE REVIEW

File: test/specs/rawr.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        HomePage.signInButton.click(); //click login link
+        // //---Login  Page
+        // //Input an email and click Create An Account Buttons
+        LoginPage.startAccountCreation(maily);
+        // //fill in your personal information
+        // //Fill in Your Address section
+        // //Press register Button
+        PersonalInfo.getPersonal();
+        // //TODO: Refactor this verification to ensure that the correct email is present in the Personal Info page
+        browser.debug();
```
Comment (DaRuski) [inline test/specs/rawr.js]: Please don't commit browser.debug() -- It will halt the automation

File: test/specs/add_Dress.js [hunk 2]
```diff
@@ -1,46 +1,29 @@
...
+        Payment.paymentChoice();
+        Payment.confirmOrder();
+        var orderConfirmed = browser.getTitle();
+        assert.equal( orderConfirmed , 'Order confirmation - My Store');
 
-        
 
     });
 });
+//var totPrice = browser.element('#total_price').getText();
```
Comment (DaRuski) [inline test/specs/add_Dress.js]: Do not check in commented code

File: test/specs/add_Dress.js [hunk 3]
```diff
@@ -1,46 +1,29 @@
...
-        browser.element('[data-id-product="3"]').click();
-        browser.pause(5000);
-        browser.element('[title="Proceed to checkout"]').click();
-        var totPrice = browser.element('#total_price').getValue();
-        assert.equal(totPrice, "$28.00");
+        HomePage.locateAndSelect();
+        CheckOut.checkOutButton.click();
+        SignIn.signInAsUser();
+        ConfirmAddress.addressConfirmation();
+        AgreeToTerms.selectCheckbox();
```
Comment (DaRuski) [inline test/specs/add_Dress.js]: There is a page named "Agree to Terms"?

File: test/specs/add_Dress.js [hunk 4]
```diff
@@ -1,46 +1,29 @@
...
-        browser.moveToObject('[title="Printed Dress"].replace-2x');
-        browser.element('[data-id-product="3"]').click();
-        browser.pause(5000);
-        browser.element('[title="Proceed to checkout"]').click();
-        var totPrice = browser.element('#total_price').getValue();
-        assert.equal(totPrice, "$28.00");
+        HomePage.locateAndSelect();
+        CheckOut.checkOutButton.click();
+        SignIn.signInAsUser();
+        ConfirmAddress.addressConfirmation();
```
Comment (DaRuski) [inline test/specs/add_Dress.js]: See below

File: test/specs/add_Dress.js [hunk 5]
```diff
@@ -1,46 +1,29 @@
...
+import AgreeToTerms from '../../page_objects/shipping.js'
+import ConfirmAddress from '../../page_objects/confirmAddress.js'
+import Payment from '../../page_objects/paymentPage.js'
 var assert = require('assert');
-var maily = Math.random() + 'testasaurusrex@dino.com';
+
 
 
 describe('webdriver.io page', function() {
     it('adding a new dress and checking out', function () {
```
Comment (DaRuski) [inline test/specs/add_Dress.js]: The title should reflect the same style as a test case

File: test/specs/add_Dress.js [hunk 6]
```diff
@@ -1,46 +1,29 @@
...
+import SignIn from '../../page_objects/signIn.js';
+import AgreeToTerms from '../../page_objects/shipping.js'
+import ConfirmAddress from '../../page_objects/confirmAddress.js'
+import Payment from '../../page_objects/paymentPage.js'
 var assert = require('assert');
-var maily = Math.random() + 'testasaurusrex@dino.com';
+
 
 
 describe('webdriver.io page', function() {
```
Comment (DaRuski) [inline test/specs/add_Dress.js]: The suite is named 'webdriver.io page'?

File: test/specs/rawr.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
+var maily = Math.random() + 'testasaurusrex@dino.com';
```
Comment (DaRuski) [inline test/specs/rawr.js]: I'm not quite sure why this file is named 'rawr'?

File: page_objects/signIn.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+class SignIn extends Page{
+
+    get mailer() { return browser.element("#email")}
+    get passy() { return browser.element('#passwd')}
+    get submitLogIn() { return browser.element('#SubmitLogin')}
+
+    signInAsUser(){
+
+        this.mailer.setValue('GAMman@Booyash.com');
+        this.passy.setValue('Epic1234');
```
Comment (DaRuski) [inline page_objects/signIn.js]: what is 'passy'?

File: page_objects/signIn.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+class SignIn extends Page{
+
+    get mailer() { return browser.element("#email")}
+    get passy() { return browser.element('#passwd')}
+    get submitLogIn() { return browser.element('#SubmitLogin')}
+
+    signInAsUser(){
+
+        this.mailer.setValue('GAMman@Booyash.com');
```
Comment (DaRuski) [inline page_objects/signIn.js]: what is 'mailer'?

File: page_objects/shipping.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import Page from './page'
+
+class Shipping extends Page{
+
+    get checkIt() { return browser.element('#cgv')}
+    get checkOut() { return browser.element('button.standard-checkout')}
+
+    selectCheckbox(){
+
+        this.checkIt.click();
```
Comment (DaRuski) [inline page_objects/shipping.js]: Please rename these

File: page_objects/paymentPage.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import Page from './page'
+
+class Payment extends Page{
+
+    get bankWire() { return browser.element('[title="Pay by bank wire"]')}
+    get confirmOrderButton() { return browser.element('button.button-medium')}
+    paymentChoice(){
```
Comment (DaRuski) [inline page_objects/paymentPage.js]: If there is a choice, according to the name of the function, why does this function not have any way to make that 'choice'?

File: page_objects/orderConfirmationPage.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import Page from './page'
+
+class OrederConfirmation extends Page{
+
+    get mailer() { return browser.element("#email")}
+    get passy() { return browser.element('#passwd')}
```
Comment (DaRuski) [inline page_objects/orderConfirmationPage.js]: Haven't these declarations been used elsewhere?

File: page_objects/orderConfirmationPage.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+    get submitLogIn() { return browser.element('#SubmitLogin')}
+
+    signInAsUser(){
```
Comment (DaRuski) [inline page_objects/orderConfirmationPage.js]: Please re-do this function to accept an email and password as a parameter.

File: test/specs/add_Dress.js (hunk 4 above)
Comment (DaRuski) [reply test/specs/add_Dress.js]: A.K.A. -- Is there a page named Confirm Address?

File: page_objects/home.page.js [hunk 14]
```diff
@@ -5,10 +5,21 @@ class HomePage extends Page {
...
 	get signInButton () { return browser.element('#header a.login') }
-
+	get orangeDress() { return browser.element('[title="Printed Dress"].replace-2x')}
+	get selection() { return browser.element('[data-id-product="3"]')}
+	get proceed()   { return browser.element('[title="Proceed to checkout"]')}
 	/**
 	 * define or overwrite page methods
 	 */
+	locateAndSelect(){
+		browser.pause(5000);
```
Comment (DaRuski) [inline page_objects/home.page.js]: Find an easier way to replace this pause by making the computer look for something instead.

File: page_objects/paymentPage.js [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+
+        this.bankWire.click();
+    }
+    confirmOrder(){
```
Comment (DaRuski) [inline page_objects/paymentPage.js]: Is it really necessary to make a _whole_ function just to do one thing?

File: test/specs/rawr.js [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+import PersonalInfo from '../../page_objects/personalInformation.js';
+import goneHome from '../../page_objects/checkOut.js'
+
+describe('Automation Practice', () => {
+
+    it('Create new Account, add item, and checkout', () => {
+        //Visit Homepage
+        HomePage.open();
+        // //Click on Sign In button
+        // browser.debug();
```
Comment (DaRuski) [inline test/specs/rawr.js]: Please do not check in commented code

File: testFile [hunk 17]
```diff
@@ -0,0 +1 @@
+I AM TESTING!!!!
```
Comment (DaRuski) [inline testFile]: I AM REVIEWING YOUR WORK!!!!
remove this file.

File: test/specs/rawr.js (hunk 1 above)
Comment (GMart818) [reply test/specs/rawr.js]: O

File: test/specs/add_Dress.js (hunk 2 above)
Comment (GMart818) [reply test/specs/add_Dress.js]: G

File: test/specs/add_Dress.js (hunk 3 above)
Comment (GMart818) [reply test/specs/add_Dress.js]: O

File: page_objects/shipping.js (hunk 10 above)
Comment (alenqa) [reply page_objects/shipping.js]: Changed checkIt to checkBox

File: page_objects/paymentPage.js (hunk 11 above)
Comment (alenqa) [reply page_objects/paymentPage.js]: Changed paymentChoice to selectBankWire in the "paymentPage.js"

File: test/specs/rawr.js (hunk 1 above)
Comment (DaRuski) [reply test/specs/rawr.js]: √

File: page_objects/shipping.js (hunk 10 above)
Comment (DaRuski) [reply page_objects/shipping.js]: √

File: page_objects/paymentPage.js (hunk 11 above)
Comment (DaRuski) [reply page_objects/paymentPage.js]: √

File: page_objects/orderConfirmationPage.js (hunk 13 above)
Comment (DaRuski) [reply page_objects/orderConfirmationPage.js]: √

File: test/specs/add_Dress.js (hunk 5 above)
Comment (DaRuski) [reply test/specs/add_Dress.js]: Try again

File: test/specs/add_Dress.js (hunk 6 above)
Comment (DaRuski) [reply test/specs/add_Dress.js]: √√

File: test/specs/rawr.js (hunk 7 above)
Comment (DaRuski) [reply test/specs/rawr.js]: √

File: page_objects/signIn.js (hunk 8 above)
Comment (DaRuski) [reply page_objects/signIn.js]: √

File: page_objects/signIn.js (hunk 9 above)
Comment (DaRuski) [reply page_objects/signIn.js]: √√√

File: page_objects/orderConfirmationPage.js (hunk 12 above)
Comment (DaRuski) [reply page_objects/orderConfirmationPage.js]: Let's try this one again.

File: page_objects/home.page.js (hunk 14 above)
Comment (DaRuski) [reply page_objects/home.page.js]: √

File: test/specs/rawr.js (hunk 16 above)
Comment (DaRuski) [reply test/specs/rawr.js]: Please retry.

File: testFile (hunk 17 above)
Comment (DaRuski) [reply testFile]: Retry pls.

**Pull request merged**